src/util/base58.rs

```diff
@@ -46,6 +46,8 @@ pub enum Error {
     TooShort(usize),
     /// Secp256k1 error while parsing a secret key
     Secp256k1(secp256k1::Error),
+    /// Hex decoding error
+    Hex(hex::Error)
```

Comment:
Great points! The question is do we want to try to do it in this PR or as a part of error handling refactor? I don't see a problem with either approach but if we're going to accept this PR it should probably *not* go to the Taproot release because we'd be removing these variants in the next version.

I believe removing `From<key::Error>` is needed. There's no reason why base58 should have anything to do with key errors.

Also would love to see `base58` as a separate crate with code quality similar to the `base64` crate.

Reply:
I had tried to keep the errors in other modules as untouched as possible in this PR but some variants had to be introduced as part of existing implementations. I hadn't actually taken a good look at where the errors could originate from. I will definitely take some time to look at it.

However I think it shouldn't be part of this PR since that changes could be pretty large.

Reply:
Yeah, I'm also a bit more in favor of not in this PR.

Reply:
> Also would love to see base58 as a separate crate with code quality similar to the base64 crate.

The base58 crate smash is super easy, I've got it queued up in my tree waiting for the correct time to push.

Reply:
I ACKed the change but next time I think it'd be better to use GH issues for that to not waste time pushing and reviewing.

Reply:
> I've got it queued up in my tree

IDK what's your design but I was thinking of also redesigning it slightly to separate encoding and checksum computation so that one can decode the bytes without being forced to compute checksum.

Reply:
Fine with having a separate `base58` crate but recall we were thinking about having `bitcoin-str` which would combine base58, hex, `rust-bitcoin-bech32` and maybe even `base64-compat`.

Anyway this discussion goes well beyond this PR :)